### PR TITLE
Add profiles base

### PR DESCRIPTION
**Pull request by khewonc**
### What does this PR do?

Add base for DAP CRD and controller

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

n/a

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: controllers/datadogagent/controller_reconcile_v2.go [hunk 1]
```diff
@@ -96,7 +96,9 @@ func (r *Reconciler) reconcileInstanceV2(ctx context.Context, logger logr.Logger
 
 	features, requiredComponents := feature.BuildFeatures(instance, reconcilerOptionsToFeatureOptions(&r.options, logger))
 	// update list of enabled features for metrics forwarder
-	r.updateMetricsForwardersFeatures(instance, features)
+	if r.options.OperatorMetricsEnabled {
```
Comment (davidor) [inline controllers/datadogagent/controller_reconcile_v2.go:99]: Is this related to the profiles?

File: controllers/datadogagentprofile/suite_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```
Comment (davidor) [inline controllers/datadogagentprofile/suite_test.go:1]: The CI is failing because the tests on this file are failing.

I think it's because they should be tagged as integration tests.

This file doesn't add anything useful for now, so I think we can delete it. We'll add proper tests later when we start implementing the feature.

File: config/crd/bases/v1/datadoghq.com_datadogagentprofiles.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                                        x-kubernetes-int-or-string: true
+                                      description: 'Requests describes the minimum amount of compute resources required. If Requests is omitted for a container, it defaults to Limits if that is explicitly specified, otherwise to an implementation-defined value. More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/'
+                                      type: object
+                                  type: object
+                              type: object
+                            type: object
+                        type: object
+                      type: object
+                  type: object
+                dapAffinity:
```
Comment (davidor) [inline config/crd/bases/v1/datadoghq.com_datadogagentprofiles.yaml:72]: It seems that this file was not update after the field was renamed.

File: config/crd/bases/v1/datadoghq.com_datadogagentprofiles.yaml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                  properties:
+                    dapNodeAffinity:
```
Comment (davidor) [inline config/crd/bases/v1/datadoghq.com_datadogagentprofiles.yaml:74]: Same for this one.

File: config/crd/bases/v1beta1/datadoghq.com_datadogagentprofiles.yaml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                                    x-kubernetes-int-or-string: true
+                                  description: 'Requests describes the minimum amount of compute resources required. If Requests is omitted for a container, it defaults to Limits if that is explicitly specified, otherwise to an implementation-defined value. More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/'
+                                  type: object
+                              type: object
+                          type: object
+                        type: object
+                    type: object
+                  type: object
+              type: object
+            dapAffinity:
```
Comment (davidor) [inline config/crd/bases/v1beta1/datadoghq.com_datadogagentprofiles.yaml:73]: This file should also be regenerated to include the rename of the field.

**Review by fanny-jiang — APPROVED** [review]
🎉

**Review by davidor — APPROVED** [review]
